tests/undernames.test.ts

```diff
@@ -0,0 +1,185 @@
+          -1,
+          -1000,
+        ])(
+          'should throw an error when an invalid quantity is provided: %s',
+          async (badQty) => {
+            const undernameInput = {
+              name: 'apple',
+              qty: badQty,
+            };
+            const writeInteraction = await contract.writeInteraction(
+              {
+                function: 'increaseUndernames',
+                ...undernameInput,
+              },
+              {
+                disableBundling: true,
+              },
+            );
+
+            expect(writeInteraction?.originalTxId).not.toBe(undefined);
+            const { cachedValue } = await contract.readState();
+            expect(Object.keys(cachedValue.errorMessages)).toContain(
+              writeInteraction!.originalTxId,
+            );
+            expect(
+              cachedValue.errorMessages[writeInteraction!.originalTxId],
+            ).toEqual(expect.stringContaining(INVALID_INPUT_MESSAGE));
+          },
+        );
+      });
+
+      describe('with valid input', () => {
+        it.each([
+          1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 100, 1000, 10000, 100000, 1000000,
+        ])(
+          'should successfully increase undernames with valid quantity provided: : %s',
+          async (goodQty) => {
+            const undernameInput = {
+              name: 'name1',
+              qty: goodQty,
+            };
+            const writeInteraction = await contract.writeInteraction(
+              {
+                function: 'increaseUndernames',
+                ...undernameInput,
+              },
+              {
+                disableBundling: true,
+              },
+            );
+
+            expect(writeInteraction?.originalTxId).not.toBe(undefined);
+            const { cachedValue } = await contract.readState();
+
+            expect(Object.keys(cachedValue.errorMessages)).not.toContain(
+              writeInteraction!.originalTxId,
+            );
+
+            // Add any additional expectations for successful execution here
+          },
+        );
+
+        it.each(['name1', 'name2', 'name3'])(
+          'should successfully increase undernames with valid name provided: : %s',
+          async (validName) => {
+            const undernameInput = {
+              name: validName,
+              qty: 1,
+            };
+            const writeInteraction = await contract.writeInteraction(
+              {
+                function: 'increaseUndernames',
+                ...undernameInput,
+              },
+              {
+                disableBundling: true,
+              },
+            );
+
+            expect(writeInteraction?.originalTxId).not.toBe(undefined);
+            const { cachedValue } = await contract.readState();
+            expect(Object.keys(cachedValue.errorMessages)).not.toContain(
+              writeInteraction!.originalTxId,
+            );
+            // Add any additional expectations for successful execution here
```

Comment:
we should check the record field and validate that the `undernames` field was properly set to the new one

Reply:
add check to see if it updated the undername count on records for that name

Reply:
added